source/slang/slang-emit-spirv.cpp

```diff
@@ -3441,15 +3452,16 @@ struct SPIRVEmitContext
                         id,
                         fieldNameDecor->getName());
                 }
-                else if (as<IRGloballyCoherentDecoration>(decor))
+                else if (auto collection = as<IRMemoryQualifierSetDecoration>(decor))
                 {
-                    emitOpMemberDecorate(
-                        getSection(SpvLogicalSectionID::Annotations),
-                        decor,
-                        spvStructID,
-                        SpvLiteralInteger::from32(id),
-                        SpvDecorationCoherent
-                    );
+                    if(collection->getMemoryQualifierBit() | MemoryQualifierSetModifier::Flags::kCoherent)
```

Comment:
&

Reply:
Thank you for catching that big mistake which I included everywhere...
I will fix it in the other areas as well.

Reply:
No problems, it is the kind of mistake I make all the time. Wonder why the test case didn't catch it.

Reply:
Maybe good to have some CHECK lines and CHECK-NOT lines to make sure we have all the right decorations.

Reply:
The tests should be checking for the modifier syntax outputted, I realized I did not add it, which I will add.